typescript-loader/fixtures/index-asserted.js

```diff
@@ -0,0 +1 @@
+import './app.js' assert { type: 'typescript' };
```

Comment:
I would think that if the user is going to write non-runnable code, it should be `import './app.ts'`, rather than an invalid assertion.

Reply:
Both options are provided so they can see the pros and cons. I would say the import assertion is in the spirit of their intended purpose and does the job quite well, without creating a fight with typescript itself.

Reply:
I personally went the `.ts` route and use esbuild for transpilation (and only bother enforcing type-checking as a PR check—VS Code does enough during dev).

Reply:
I don’t think we should be pushing the assertion approach. The assertion is a _validation_ step; it’s not supposed to inform the runtime _how_ to interpret what the runtime is importing. The use here of the assertion as a signal to the runtime/loader of what to do goes against that principle, and could lead to confusion in users of what the assertion is meant for.

Reply:
I thought of that: but in a roundabout way, it does kind of does align with a validation—if it's wrong, it still blows up.

Reply:
What if there are two files, one named `app.js`, one named `app.ts`? Would you throw an exception? Would you discard `app.js` and transpile `app.ts`? Would you load `app.js` and ignore the import assertion?

---

Personally I'm very frustrated by the choice of TypeScript not to allow to refer TS modules using their actual extension, I think Deno stance on it makes much more sense, and I'd prefer if this loader was encouraging to write code that can work on Deno rather than trying to come up with a new convention.

Reply:
Ooow damn. Good point @aduh95!

Ugh, yes me too. I gotta stop myself before I go on a rant 🙊 

I dunno how Deno does it; I'll take a look 🙂

Reply:
> I think Deno stance on it makes much more sense

Link?

Reply:
I don't know a good link to it, but basically, it requires the file extension to be supplied in the import specifier, and doesn't magically load `tsFile.ts` when doing `import './tsFile.js';` – it would simply crash, you must `import './tsFile.ts';` instead, or have an import maps that link to the actual file. It's explained briefly in https://deno.land/manual/typescript/types#types-and-type-declarations.

Surcase and ESBuild take a more lenient approach by allowing both `.ts` and `.js` when referencing TS files (so effectively supporting both Deno-specific and tsc-specific TS files).